docs/whats-new/preview-5.md

```diff
@@ -0,0 +1,595 @@
+the following settings to user `secrets.json` file:
+
+```json
+{
+  "Azure": {
+    "SubscriptionId": "<subscription id>",
+    "Location": "<default location>",
+    "ResourceGroup": "<resource group name>"
+  }
+}
+```
+
+When you launch the AppHost the dashboard will show that it is create the Azure resources for you and provide helpful links to the deployment into the Azure portal, or in the case of failure logs that provide hints as to what might be causing the deployment issue.
+
+:::image type="content" source="/media/preview-5/azure-resource-provisioning-on-dashboard.png" lightbox="/media/preview-5/azure-resource-provisioning-on-dashboard.png" alt-text=".NET Aspire dashboard: Azure provisioning.":::
+
+To use Azure provisioning you only need to make use of one of an Azure resource. For example:
+
+```csharp
+var builder = DistributedApplication.CreateBuilder(args);
+
+var blobs = builder.AzureStorage("storage")
+                   .AddBlobs("blobs");
+
+builder.AddProject<Projects.GalleryApp>("galleryapp")
+       .WithReference(blobs);
+```
+
+Some resources in .NET Aspire such as Postgres are not cloud specific and can run locally in a container but when deployed use a managed service (such as Azure Postgres Flexible Server).
+
+```csharp
+var builder = DistributedApplication.CreateBuilder(args);
+
+var db = builder.AddPostgres("pgsql")
+                .PublishAsAzurePostgresFlexibleServer()
+                .AddDatabase("inventorydb");
+
+builder.AddProject<Projects.InventoryApi>("inventoryapi")
+       .WithReference(db);
+```
+
+It is also possible to use an Azure hosted resource for local development by using the `AsAzurePostgresFlexibleServer()` extension method instead. When this method is used the container will not be started locally and cloud-based instance will be created just like the Azure only resource types. The `PublishAsX` and `AsX` methods also support callbacks to customize the underlying Azure resources as shown above in the Cosmos DB example.
+
+For more information, see [Local Azure provisioning](../deployment/azure/local-provisioning.md).
+
+## Other Azure-related improvements
+
+There are several other Azure-related improvements in preview 5, the following sections outline these improvements.
+
+### No more exposed endpoint selection in AZD
+
+In previous previews, when an Aspire application was deployed to Azure, part of the initialization
+process for Azure Developer CLI (`azd`) was to select the services that would be exposed externally
+to the Internet. This included container-based resources that may or may not be hardened for Internet
+access.
+
+In preview 5 we have worked with the Azure Developer CLI team to not display this prompt. Now, by default
+all endpoints exposed container resources are exposed only via the internal Azure Container Apps endpoint. To expose endpoints on a container resource you need to use the `WithExternalHttpEndpoints(...)` extension
```

Comment:
It's not just container resources, this is all endpoints correct @davidfowl?

Reply:
Correct